### ISSUE TITLE
[feature] all_characters_matched_by()

### ISSUE THREAD

**Issue by jidanni** [Perl/perl5#18386]
On perlrecharclass,
```
       A "\w" matches a single alphanumeric character (an alphabetic....
```
And then it goes on and on, for hundreds of words, discussing all kinds
of exceptions, almost like, "If your grandmother isn't home, then `\w`
won't match zzz to yyy" etc.

Same for
```
       "\s" matches any single character considered whitespace.
```
OK. I have an idea:
Have a command
```
print all_characters_matched_by(/\w/);
```
Which would print `ABCDE`...`89` or whatever.
One could also do
```
all_characters_matched_by(/\w/a);
all_characters_matched_by(/\s/a); #etc.
```

Yup! No more pouring over the manuals in the middle of the night,
and even then not being entirely sure.

Just temporarily stick all_characters_matched_by() into your program,
and get the exact list right from perl itself. And it will already know
if your grandmother is home or it is sunny outside... all the
`$ENVironmental` factors affecting the result considered. No static man
page could beat that.

**Comment by jidanni** [comment]
Or if such a thing already exists, perlrecharclass could mention it.

**Comment by Grinnz** [comment]
I certainly think such a thing could be useful, but implementation and syntax seems very tricky. For example, you are providing regexes, but what we really want to know is what a character class includes rather than allowing arbitrary regexes. Similar to the problem with trim in #17952 Perl has no way to specify a character class as a first-class value, and specifying it as a string has a high probability of user error.

**Comment by jidanni** [comment]
OK. Maybe just add to the document:

```
If you insist on knowing all characters that match e.g., "\w",
you can simply use
$ perl -C -we 'for(0..0xFFFFFF){$_ = chr; print if /\w/;}'
```

**Comment by khwilliamson** [comment]
On 12/6/20 5:18 PM, 積丹尼 Dan Jacobson wrote:
> OK. Maybe just add to the document:
> 
> |If you insist on knowing all characters that match e.g., "\w", you can 
> simply use $ perl -C -we 'for(0..0xFFFFFF){$_ = chr; print if /\w/;}' |
> 
> —
>
Using -Dr or use re qw(Debug COMPILE) will give the lowest code points 
matched.  There is an environment variable you can set (I have to look 
it up each time by grepping regcomp.c) that you can set to a large 
number so that it will output all the code points.  Then you still would 
have to parse that output if you want to know what characters they are.

BTW nothing above 0x10FFFF will ever match any such property.

**Comment by jidanni** [comment]
> Using -Dr or use re ...

That is all great but recipes need to be provided for the average user.

Indeed, just showing
```
$ perl -e 'for(0..0x7F){$_ = chr; print if /\w/;}'
0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz
```
is enough to get folks started... (But needs -C and the higher limit...)

**Comment by khwilliamson** [comment]
On 12/6/20 11:34 PM, 積丹尼 Dan Jacobson wrote:
>     Using -Dr or use re ...
> 
> That is all great but recipes need to be provided for the average user.
> 
> Indeed, just showing
> 
> |$ perl -e 'for(0..0x7F){$_ = chr; print if /\w/;}' 
> 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz |
> 
> is enough to get folks started...
> 

Well, actually that info is in perluniprops.pod, and you can get a list 
of all code points that match a given property by using 
Unicode::UCD::prop_invlist().  There is a recipe in the documentation 
for that module.

You need to know what \w is equivalent to.  That information is found by 
starting at any of the regular expression pods.

GIven this, perhaps just a sentence or two in various places would suffice.

I suggest that you read perlrebackslash and perlrecharclass and come 
back with what more you think is needed

**Comment by jidanni** [comment]
Yes, on perlrecharclass at the end of the mile long "Word characters"
section, and perhaps even on perlrebackslash, after

> The exact set of characters matched by "\d", "\s", and "\w" varies
> depending on various pragma and regular expression modifiers.

You folks could add

"The exact list of characters matched by "\w" at any moments is not in
fact a National Security Secret, and can be simply printed out by [my
brilliant one-liner]."

perluniprops.pod, Unicode::UCD::prop_invlist() are interesting, but the
moment anybody starts mentioning them on man pages, one goes into a
never ending tailspin of making this more and more complicated.
So any person mentioning them needs to control themselves to not go
beyond a couple words, like "see also."

> You need to know what \w is equivalent to. That information is found by
> starting at any of the regular expression pods.

Well yes, but my whole point is the reader has already got too tangled
up reading and just wants to spit out the actual raw character list.
